src/main/java/org/testng/internal/MethodInvocationHelper.java

```diff
@@ -291,9 +292,19 @@ private static void invokeWithTimeoutWithNoExecutor(
         new InvokeMethodRunnable(tm, instance, parameterValues, hookable, testResult);
     long startTime = System.currentTimeMillis();
     long realTimeOut = MethodHelper.calculateTimeOut(tm);
+    boolean notTimedout = true;
     try {
+      Thread currentThread = Thread.currentThread();
+      new Thread(() -> {
+        try {
+          TimeUnit.MILLISECONDS.sleep(realTimeOut);
+        } catch (InterruptedException e) {
+        }

```

Comment:
I wonder if we should add `Thread.currentThread().interrupt();`
https://stackoverflow.com/questions/1087475/when-does-javas-thread-sleep-throw-interruptedexception